src/entity/ReadingStreakActions.ts

```diff
@@ -0,0 +1,24 @@
+import { Column, Entity, OneToOne, PrimaryColumn } from 'typeorm';
+import { UserStreak } from './user';
+
+export enum ReadingStreakActionType {
+  Recover = 'recovered',
+}
+
+@Entity()
+export class ReadingStreakActions {
+  @PrimaryColumn({ type: 'text' })
+  id: string;
+
+  @OneToOne(() => UserStreak, (userStreak) => userStreak.userId, {
+    lazy: true,
+    onDelete: 'CASCADE',
+  })
+  userStreak: Promise<UserStreak>;
+
+  @Column({ type: 'timestamptz', default: () => 'now()' })
+  timestamp: Date;
+
+  @Column({ type: 'text' })
+  type: string;
```

Comment:
typescript type here can still be `ReadingStreakActionType` even though in psql its `text`

Reply:
see `UserAction` for examples